src/functions/module_statistical.ts

```diff
@@ -579,6 +645,62 @@ export const LARGE: AddFunctionDescription = {
   isExported: true,
 };
 
+// -----------------------------------------------------------------------------
+// MATTHEWS
+// -----------------------------------------------------------------------------
+export const MATTHEWS: AddFunctionDescription = {
+  description: "Compute the Spearman rank correlation coefficient of a dataset.",
+  args: [
+    arg(
+      "data_x (range<number>)",
+      _lt("The range representing the array or matrix of observed data.")
+    ),
+    arg(
+      "data_y (range<number>)",
+      _lt("The range representing the array or matrix of predicted data.")
+    ),
+  ],
+  returns: ["NUMBER"],
+  compute: function (X: MatrixArgValue, Y: MatrixArgValue): number {
+    const flatX = X.flat();
+    const flatY = Y.flat();
+    if (flatX.length !== flatY.length) {
+      throw new Error("The two arrays must have the same length");
```

Comment:
this should be translated.  I think you could make some generic assert functions like we did for the financial functions for example to avoid duplicating stuff